pkg/types/azure/validation/featuregates.go

```diff
@@ -0,0 +1,38 @@
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	features "github.com/openshift/api/features"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/featuregates"
+)
+
+// GatedFeatures determines all of the install config fields that should
+// be validated to ensure that the proper featuregate is enabled when the field is used.
+func GatedFeatures(c *types.InstallConfig) []featuregates.GatedInstallConfigFeature {
+	cp := c.ControlPlane.Platform
+	defMp := c.Platform.Azure.DefaultMachinePlatform
+	return []featuregates.GatedInstallConfigFeature{
+		{
+			FeatureGateName: features.FeatureGateMachineAPIMigration,
+			Condition:       cp.Azure != nil && cp.Azure.Identity != nil && cp.Azure.Identity.SystemAssignedIdentityRole != nil,
```

Comment:
We can specify identity.type as `SystemAssigned` without explicit `systemAssignedIdentityRole`, then it defaults to `Contributor` role.
```
controlPlane:
  platform:
    azure:
      identity:
        type: SystemAssigned
```
Installation with such configuration can continue without featuregate enabled, does this also need to be restricted and only run when featuregate is enabled?

Reply:
Correct. I have fixed it so that we check on the Type. I also added some static validations to restrict this further. Should be set.